frontends/p4/def_use.h

```diff
@@ -332,16 +336,24 @@ struct hash<P4::ProgramPoint> {
 };
 }  // namespace std
 
+namespace Util {
+template <>
+struct Hasher<P4::ProgramPoint> {
+    size_t operator()(const P4::ProgramPoint &p) const { return p.hash(); }
+};
+}  // namespace Util
+
 namespace P4 {
 class ProgramPoints : public IHasDbPrint {
-    typedef std::unordered_set<ProgramPoint> Points;
+    typedef absl::flat_hash_set<ProgramPoint, Util::Hash> Points;
```

Comment:
If we want to do this in most places where we can, it might be worth defining a using in `lib/map.h`, something like `template<typename T> using FlatHashSet = absl::flat_hash_set<T, Util::Hash>`.

Reply:
This would require some massive changes to all dependencies and increased include size. So I'd rather keep it fine-grained for now. Otherwise we may end with the same proliferation as we have now with `ordered_map` (in many cases w/o particular reason)